docs/source/export.rst

```diff
@@ -0,0 +1,65 @@
+Export
+======
+
+Export is a new feature in HDMF 2.0. You can use export to take a container that was read from a file and write it to
+a different file, with or without modifications to the container in memory.
+The in-memory container being exported will be written to the exported file as if it was never read from a file.
+
+FAQ
+---
+
+Can I read a container from disk, modify it, and then export the modified container?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Yes, you can export the in-memory container after modifying it in memory. The modifications will appear in the exported
+file and not the read file.
+
+- If the modifications are removals or additions of containers, then no special action must be taken, as long as the
+  container hierarchy is updated correspondingly.
+- If the modifications are changes to attributes, then
+  :py:meth:`Container.set_modified() <hdmf.container.AbstractContainer.set_modified>` must be called
+  on the container before exporting.
+
+.. note::
+
+  Modifications to :py:class:`h5py.Dataset <h5py.Dataset>` objects act *directly* on the read file on disk.
+  Changes are applied immediately and do not require exporting or writing the file.
```

Comment:
```suggestion
  Changes are applied immediately and do not require exporting or writing the file. If you want to modify a dataset only in the new file, than you should replace the whole object with a new array holding the modified data. To prevent unintentional changes to the source file, the source file should be opened with ``mode='r'``.
```